src/main/java/net/minecraftforge/items/wrapper/ShulkerItemStackInvWrapper.java

```diff
@@ -0,0 +1,239 @@
+                itemStacks.set(slot, reachedLimit ? ItemHandlerHelper.copyStackWithSize(stack, limit) : stack);
+            }
+            else
+            {
+                existing.grow(reachedLimit ? limit : stack.getCount());
+            }
+            setItemList(itemStacks);
+        }
+
+        return reachedLimit ? ItemHandlerHelper.copyStackWithSize(stack, stack.getCount()- limit) : ItemStack.EMPTY;
+    }
+
+    @Override
+    @NotNull
+    public ItemStack extractItem(int slot, int amount, boolean simulate)
+    {
+        NonNullList<ItemStack> itemStacks = getItemList();
+        if (amount == 0)
+            return ItemStack.EMPTY;
+
+        validateSlotIndex(slot);
+
+        ItemStack existing = itemStacks.get(slot);
+
+        if (existing.isEmpty())
+            return ItemStack.EMPTY;
+
+        int toExtract = Math.min(amount, existing.getMaxStackSize());
+
+        if (existing.getCount() <= toExtract)
+        {
+            if (!simulate)
+            {
+                itemStacks.set(slot, ItemStack.EMPTY);
+                setItemList(itemStacks);
+                return existing;
+            }
+            else
+            {
+                return existing.copy();
+            }
+        }
+        else
+        {
+            if (!simulate)
+            {
+                itemStacks.set(slot, ItemHandlerHelper.copyStackWithSize(existing, existing.getCount() - toExtract));
+                setItemList(itemStacks);
+            }
+
+            return ItemHandlerHelper.copyStackWithSize(existing, toExtract);
+        }
+    }
+
+    protected void validateSlotIndex(int slot)
+    {
+        if (slot < 0 || slot >= getSlots())
+            throw new RuntimeException("Slot " + slot + " not in valid range - [0," + getSlots() + ")");
+    }
+
+    @Override
+    public int getSlotLimit(int slot)
+    {
+        return 64;
+    }
+
+    @Override
+    public boolean isItemValid(int slot, @NotNull ItemStack stack)
+    {
+        return stack.getItem().canFitInsideContainerItems();
+    }
+
+    @Override
+    public void setStackInSlot(int slot, @NotNull ItemStack stack) {
+        validateSlotIndex(slot);
+        if (!isItemValid(slot, stack)) throw new RuntimeException("Invalid stack " + stack + " for slot " + slot + ")");
+        NonNullList<ItemStack> itemStacks = getItemList();
+        itemStacks.set(slot, stack);
+        setItemList(itemStacks);
+    }
+
+    protected NonNullList<ItemStack> getItemList()
+    {
+        CompoundTag rootTag = BlockItem.getBlockEntityData(this.stack);
+        if ((cachedTag != null && !cachedTag.equals(rootTag)) || rootTag == null)
+            itemStacksCache = refreshItemList(rootTag);
+        return itemStacksCache;
+    }
+
+    protected NonNullList<ItemStack> refreshItemList(CompoundTag rootTag)
+    {
+        NonNullList<ItemStack> itemStacks = NonNullList.withSize(getSlots(), ItemStack.EMPTY);
+        if (rootTag != null && rootTag.contains("Items", 9)) {
```

Comment:
- Formatting, braces on new lines
- Use `Tag.TAG_LIST` instead of the magic number